cdap-ui/app/cdap/components/Replicator/utilities/index.ts

```diff
@@ -246,6 +232,142 @@ export function constructTablesSelection(tables, columns, dmlBlacklist) {
+    name: rawConfig.label || rawConfig.name || '',
+    description: objectQuery(rawConfig, 'config', 'description') || '',
+    activeStep: 1,
+    offsetBasePath: objectQuery(rawConfig, 'config', 'offsetBasePath') || '',
+    numInstances: objectQuery(rawConfig, 'config', 'parallelism', 'numInstances') || 1,
+  };
+
+  const stages = objectQuery(rawConfig, 'config', 'stages') || [];
+
+  // SOURCE
+  const source = stages.find((stage) => {
+    const stageType = objectQuery(stage, 'plugin', 'type');
+    return stageType === PluginType.source;
+  });
+
+  if (source) {
+    const sourceArtifact = objectQuery(source, 'plugin', 'artifact') || {};
+
+    const sourcePluginInfo = await fetchPluginInfo(
+      parentArtifact,
+      sourceArtifact.name,
+      sourceArtifact.scope,
+      source.plugin.name,
+      source.plugin.type
+    ).toPromise();
+
+    newState.sourcePluginInfo = sourcePluginInfo;
+    newState.sourceConfig = objectQuery(source, 'plugin', 'properties') || {};
+
+    try {
+      const sourcePluginWidget = await fetchPluginWidget(
+        sourceArtifact.name,
+        sourceArtifact.version,
+        sourceArtifact.scope,
+        sourcePluginInfo.name,
+        sourcePluginInfo.type
+      ).toPromise();
+
+      newState.sourcePluginWidget = sourcePluginWidget;
+    } catch (e) {
+      // tslint:disable-next-line: no-console
+      console.log('Cannot fetch source plugin widget', e);
+      // no-op
+    }
+
+    if (Object.keys(newState.sourceConfig).length > 0) {
+      newState.activeStep = 2;
+    }
+  }
+
+  // TABLES & COLUMNS
+  const tables = objectQuery(rawConfig, 'config', 'tables');
+  if (tables) {
+    let selectedTables: ITablesStore = Map();
+
+    let columns: IColumnsStore = Map();
+    let dmlBlacklist: IDMLStore = Map();
+    tables.forEach((table) => {
+      const tableKey = generateTableKey(table);
+
+      const tableInfo: ITable = {
+        database: table.database,
+        table: table.table,
+      };
+
+      if (table.schema) {
+        tableInfo.schema = table.schema;
+      }
+
+      selectedTables = selectedTables.set(tableKey, fromJS(tableInfo));
+
+      const tableColumns = objectQuery(table, 'columns') || [];
+      const columnList = fromJS(tableColumns);
+
+      columns = columns.set(tableKey, columnList);
+
+      const tableDML = objectQuery(table, 'dmlBlacklist') || [];
+      dmlBlacklist = dmlBlacklist.set(tableKey, Set<DML>(tableDML));
+    });
+
+    newState.tables = selectedTables;
+    newState.columns = columns;
+    newState.dmlBlacklist = dmlBlacklist;
+
+    if (selectedTables.size > 0) {
+      newState.activeStep = 3;
+    }
+  }
+
+  // TARGET
+  const target = stages.find((stage) => {
+    const stageType = objectQuery(stage, 'plugin', 'type');
+    return stageType === PluginType.target;
+  });
+
+  if (target) {
+    const targetArtifact = objectQuery(target, 'plugin', 'artifact') || {};
+
+    const targetPluginInfo = await fetchPluginInfo(
```

Comment:
nit: Any specific reason why `fetchPluginInfo` is not wrapped in `try...catch` block but `fetchPluginWidget` is

Reply:
yeah, if the fetchPluginInfo fails, we actually want to fail this function, so the failure is getting caught by whoever is calling this function. The fetchPluginWidget can fail, but that shouldn't make this function fail, so there's a try catch block here to ignore the error